packages/contracts/contracts/L1/OptimismPortal.sol

```diff
@@ -0,0 +1,95 @@
+//SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import { DepositFeed } from "./DepositFeed.sol";
+import { WithdrawalVerifier } from "./Lib_WithdrawalVerifier.sol";
+import { L2OutputOracle } from "./L2OutputOracle.sol";
+
+contract OptimismPortal is DepositFeed {
+    event WithdrawalFinalized(bytes32 indexed);
+
+    // Value used to reset the l2Sender. This is more gas efficient that setting it to zero.
+    address internal constant DEFAULT_L2_SENDER = 0x000000000000000000000000000000000000dEaD;
+
+    uint256 public immutable FINALIZATION_WINDOW;
+    L2OutputOracle public immutable L2_ORACLE;
+
+    address public l2Sender = DEFAULT_L2_SENDER;
+    mapping(bytes32 => bool) public finalizedWithdrawals;
+
+    constructor(L2OutputOracle _l2Oracle, uint256 _finalizationWindow) {
+        L2_ORACLE = _l2Oracle;
+        FINALIZATION_WINDOW = _finalizationWindow;
+    }
+
+    /**
+     * Accepts value so that users can send ETH directly to this contract and
+     * have the funds be deposited to their address on L2.
+     * Note: this is intended as a convenience function for EOAs. Contracts should call the
+     * depositTransaction() function directly.
+     */
+    receive() external payable {
+        depositTransaction(msg.sender, msg.value, 30000, false, bytes(""));
```

Comment:
This doesn't inherit the `receive` function from `DepositFeed`?

Reply:
I picked it to cover the 21k intrinsic gas + a buffer but i'm not 100% sure that is sane, somebody more familiar with the go code should chime in

Reply:
> This doesn't inherit the receive function from DepositFeed?

I relocated it here, because... well, it felt to me like a receive function should be visible in the child contract, ie. the `DepositFeed` is the low level logic, the `OptimismPortal` is more business logic. I could go either way tho.

Reply:
True, that makes sense. Not super opinionated as both will be on etherscan when verified